c/tskit/tables.c

```diff
@@ -10333,12 +10356,33 @@ tsk_table_collection_check_tree_integrity(const tsk_table_collection_t *self)
         }
         num_trees++;
     }
+    tsk_bug_assert(j == num_edges);
+    while (k < num_edges) {
+        /* At this point it must be that used_edges[O[k]] == 1,
+         * since otherwise we would have added a different edge twice,
+         * and so hit the error above. */
+        if (edge_right[O[k]] != sequence_length) {
+            ret = TSK_ERR_TABLES_BAD_INDEXES;
+            goto out;
+        }
+        used_edges[O[k]] += 1;
+        k++;
+    }
+    for (j = 0; j < num_edges; j++) {
```

Comment:
Since this is super-perf critical I don't think we should do this check. I'm sure you've caught all the weird possibilities.